test-data-generator/src/main/scala/za/co/absa/spline/testdatagen/GraphType.scala

```diff
@@ -15,14 +15,20 @@
  */
 
 package za.co.absa.spline.testdatagen
+import scala.collection.immutable
 
-object GraphType extends Enumeration {
-  type GraphType = Value
+sealed abstract class GraphType(val value: String)
 
-  val ChainType: Value = Value("chain")
-  val DiamondType: Value = Value("diamond")
-  val TriangleType: Value = Value("triangle")
+object GraphType {
 
-  def withNameWithDefault(name: String): Value =
-    values.find(_.toString.toLowerCase() == name.toLowerCase()).getOrElse(ChainType)
-}
+  case object ChainType extends GraphType("chain")
+  case object DiamondType extends GraphType("diamond")
+  case object TriangleType extends GraphType("triangle")
```

Comment:
```suggestion
  case object ChainType extends GraphType("chain")

  case object DiamondType extends GraphType("diamond")

  case object TriangleType extends GraphType("triangle")
```
nit: we separate class definitions with a blank line